packages/config/test/project/utils.spec.ts

```diff
@@ -0,0 +1,1182 @@
+      it('returns path of file relative to projectRoot', () => {
+        const projectRoot = `\\root\\projects`
+        const supportFile = `\\root\\projects\\cypress\\support\\e2e.js`
+
+        expect(relativeToProjectRoot(projectRoot, supportFile)).to.eq(`cypress\\support\\e2e.js`)
+      })
+    })
+  })
+
+  context('.setSupportFileAndFolder', () => {
+    it('does nothing if supportFile is falsey', () => {
+      const obj = {
+        projectRoot: '/_test-output/path/to/project',
+      }
+
+      return setSupportFileAndFolder(obj, getCtx().file.getFilesByGlob)
+      .then((result) => {
+        expect(result).to.eql(obj)
+      })
+    })
+
+    it('sets the full path to the supportFile and supportFolder if it exists', () => {
+      const projectRoot = process.cwd()
+
+      const obj = setAbsolutePaths({
+        projectRoot,
+        supportFile: 'test/project/utils.spec.ts',
+      })
+
+      return setSupportFileAndFolder(obj, getCtx().file.getFilesByGlob)
+      .then((result) => {
+        expect(result).to.eql({
+          projectRoot,
+          supportFile: `${projectRoot}/test/project/utils.spec.ts`,
+          supportFolder: `${projectRoot}/test/project`,
+        })
+      })
+    })
+
+    it('sets the supportFile to default e2e.js if it does not exist, support folder does not exist, and supportFile is the default', () => {
+      const projectRoot = Fixtures.projectPath('no-scaffolding')
+
+      const obj = setAbsolutePaths({
+        projectRoot,
+        supportFile: 'cypress/support/e2e.js',
+      })
+
+      return setSupportFileAndFolder(obj, getCtx().file.getFilesByGlob)
+      .then((result) => {
+        expect(result).to.eql({
+          projectRoot,
+          supportFile: `${projectRoot}/cypress/support/e2e.js`,
+          supportFolder: `${projectRoot}/cypress/support`,
+        })
+      })
+    })
+
+    it('finds support file in project path that contains glob syntax', () => {
+      const projectRoot = Fixtures.projectPath('project-with-(glob)-[chars]')
+
+      const obj = setAbsolutePaths({
+        projectRoot,
+        supportFile: 'cypress/support/e2e.js',
+      })
+
+      return setSupportFileAndFolder(obj, getCtx().file.getFilesByGlob)
+      .then((result) => {
+        expect(result).to.eql({
+          projectRoot,
+          supportFile: `${projectRoot}/cypress/support/e2e.js`,
+          supportFolder: `${projectRoot}/cypress/support`,
+        })
+      })
+    })
+
+    it('sets the supportFile to false if it does not exist, support folder exists, and supportFile is the default', () => {
+      const projectRoot = Fixtures.projectPath('empty-folders')
+
+      const obj = setAbsolutePaths({
+        projectRoot,
+        supportFile: false,
+      })
+
+      return setSupportFileAndFolder(obj, getCtx().file.getFilesByGlob)
+      .then((result) => {
+        expect(result).to.eql({
+          projectRoot,
+          supportFile: false,
+        })
+      })
```

Comment:
You should be able to stub this out and we should not be pulling fixtures into this package.

```suggestion
       const getFilesByGlob = sinon.stub()
      return setSupportFileAndFolder(obj, getCtx().file.getFilesByGlob)
      .then((result) => {
        expect(result).to.eql({
          projectRoot,
          supportFile: false,
        })
        expect(getFilesByGlob).not.have.been.called
      })
```

Reply:
Thanks, I should have thought more about the solutions. I finally removed `data-context` from `config`.